### DISCUSSION TITLE
[HW4] Problem 2.1 & 3.1

### DISCUSSION THREAD

**Discussion by Yassawe** [seominjoon/kaist-ai605#161] [Q&A] [answered]
Hello Professor,

Sorry for the trivial question, but in problems 2.1 & 3.1 is it ok to train on a small subset of the datasets (like they've done in tutorials), instead of training on the full datasets? Training on the full datasets takes a significant amount of time on weak machines (and on collab too) :(

Thanks!

**Comment by seominjoon** [comment, marked as answer]
If it takes a week, I think you are doing something wrong. If you are training around 3 epochs, it should take a few hours at most. If it takes too much time, consider training with 1 epoch only. That being said, you won't be penalized for training with a subset as long as it achieves a reasonable accuracy (that is, EM bigger than 50%).
